torchvision/csrc/ROIAlign.h

```diff
@@ -107,7 +64,8 @@ class ROIAlignFunction : public torch::autograd::Function<ROIAlignFunction> {
     ctx->saved_data["aligned"] = aligned;
     ctx->saved_data["input_shape"] = input.sizes();
     ctx->save_for_backward({rois});
-    auto result = ROIAlign_forward(
+    at::AutoNonVariableTypeMode g;
```

Comment:
Why an explicit autograd-disabling guard here? Does `torch::autograd::Function` not disable autograd automatically around forward?

The pre-PR forward doesn't use an explicit guard, and afaik Python-side `torch.autograd.Function` does disable autograd around its forward method. Both of these lead me to expect `torch::autograd::Function` also disables autograd around forward.  (If it doesn't, I think it should.  Aligning its behavior with the Python version makes sense to me.  But that would be a Pytorch-side change.)

Reply:
https://github.com/pytorch/pytorch/issues/40736